core/types.ml

```diff
@@ -649,282 +539,305 @@ let primary_kind_of_type_arg : type_arg -> PrimaryKind.t =
 let add_quantified_vars qs vars =
   List.fold_right IntSet.add (List.map var_of_quantifier qs) vars
 
-(* unl type stuff *)
 
-let is_unl_point =
-  fun f (rec_vars, quant_vars) point ->
-    begin
-      match Unionfind.find point with
-      | `Closed -> true
-      | `Var (var, (lin, _), _) -> IntSet.mem var quant_vars || Linearity.is_nonlinear lin
-      | `Body t -> f (rec_vars, quant_vars) t
-      | `Recursive (var, t) ->
-        check_rec var rec_vars true (fun rec_vars' -> f (rec_vars', quant_vars) t)
-    end
+module type Constraint = sig
```

Comment:
In the context of reviewing this PR, it's obvious, but would you mind adding a comment describing what a `Constraint` abstracts over?